### PR TITLE
feat: provide build instructions for NVIDIA kernel module

### PR DESCRIPTION
**Pull request by smira**
Actual module is built in the CI, but never pushed to the registry.

Fixes talos-systems/talos#4692

Signed-off-by: Andrey Smirnov <andrey.smirnov@talos-systems.com>

### CODE REVIEW

File: containerd/pkg.yaml [hunk 1]
```diff
@@ -9,8 +9,8 @@ steps:
         # sync with version and revision in build
       - url: https://github.com/containerd/containerd/archive/refs/tags/v1.5.8.tar.gz
         destination: containerd.tar.gz
-        sha256: a41ab8d39393c9456941b477c33bb1b221a29b635f1c9a99523aab2f5e74f790
-        sha512: c769506ff6d98689c46ffee94d70ae00ef2f32e0daac1e631cbe8a587f67c7e4f83eb3895707362bdf46198b61823c99df1d8ca61095ab1415de5596f106fd07
+        sha256: 0890f7b0ee8e20a279a617c60686874b3c7a99e064adb2b38d884499b5284c43
```
Comment (smira) [inline containerd/pkg.yaml:12]: once again GitHub changed hash for the archive behind the scenes

File: kernel/config/certs/x509.genkey [hunk 2]
```diff
@@ -0,0 +1,17 @@
+[ req ]
+default_bits = 4096
+distinguished_name = req_distinguished_name
+prompt = no
+string_mask = utf8only
+x509_extensions = myexts
+
+[ req_distinguished_name ]
+O = Sidero Labs, Inc.
```
Comment (smira) [inline kernel/config/certs/x509.genkey:9]: this allows to generate kernel signing key with our overrides

File: kernel/config/pkg.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        cd /src
+        cp -v /pkg/config-${CARCH} .config
+        cp -v /pkg/certs/* certs/
+    build:
+      - |
+        cd /src
+        python3 /toolchain/kconfig-hardened-check/bin/kconfig-hardened-check -c .config -p ${KARCH} -m json | python3 /pkg/scripts/filter-hardened-check.py
+      - |
+        cd /src
+        make scripts certs
```
Comment (smira) [inline kernel/config/pkg.yaml:22]: this builds necessary scripts for signing files and also generates the kernel signing key

File: nonfree/kmod-nvidia/pkg.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        export GUESS_MD5_PATH=/toolchain/bin
+
+        rm -f /dev/tty && ln -s /dev/stdout /dev/tty
+        ln -s /toolchain/bin/echo /toolchain/bin/which
+
+        /toolchain/bin/bash nvidia.run --extract-only
+      - |
+        cd /src
+
+        make -j $(nproc) modules
```
Comment (smira) [inline nonfree/kmod-nvidia/pkg.yaml:32]: we have to rebuild `modules` here to generate `Module.symvers`

File: .drone.yml [hunk 5]
```diff
@@ -44,6 +44,25 @@ steps:
       - name: docker
         path: /root/.docker/buildx
 
+  - name: build-nonfree-pull-request
```
Comment (smira) [inline .drone.yml:47]: only build non-free, never push it

**Review by frezbo — APPROVED** [review]

**Review by Unix4ever — APPROVED** [review]

**Pull request merged**